deepsubtree.go

```diff
@@ -16,23 +17,134 @@ type DeepSubTree struct {
 	*MutableTree
 }
 
+type DSTNonExistenceProof struct {
+	*ics23.NonExistenceProof
+	LeftSiblingProof  *ics23.ExistenceProof
+	RightSiblingProof *ics23.ExistenceProof
+}
+
+// NewDeepSubTree returns a new deep subtree with the specified cache size, datastore, and version.
+func NewDeepSubTree(db dbm.DB, cacheSize int, skipFastStorageUpgrade bool, version int64) (*DeepSubTree, error) {
+	ndb := newNodeDB(db, cacheSize, nil)
+	head := &ImmutableTree{ndb: ndb, version: version}
+	mutableTree := &MutableTree{
+		ImmutableTree:            head,
+		lastSaved:                head.clone(),
+		orphans:                  map[string]int64{},
+		versions:                 map[int64]bool{},
+		allRootLoaded:            false,
+		unsavedFastNodeAdditions: make(map[string]*FastNode),
+		unsavedFastNodeRemovals:  make(map[string]interface{}),
+		ndb:                      ndb,
+		skipFastStorageUpgrade:   skipFastStorageUpgrade,
+	}
+	return &DeepSubTree{MutableTree: mutableTree}, nil
+}
+
+// Constructs a DSTNonExistenceProof using an ICS23 Non-Existence proof
+// and sibling node proofs. Returns the constructed DSTNonExistenceProof.
+func ConvertToDSTNonExistenceProof(
+	tree *MutableTree,
+	nonExistenceProof *ics23.NonExistenceProof,
+) (*DSTNonExistenceProof, error) {
+	dstNonExistenceProof := DSTNonExistenceProof{
+		NonExistenceProof: nonExistenceProof,
+	}
+	if nonExistenceProof.Left != nil {
+		leftSibling, err := tree.getSiblingNode(nonExistenceProof.Left.Key)
```

Comment:
Why do you need the `getSiblingNode` method?  Can't you directly call `recursiveGetSiblingNode`?

Reply:
Yes, I could but I find it cleaner to have a wrapper method :)

Reply:
Turned `getSiblingNode` into an exported method. Cleaner to have a wrapper method with fewer arguments and stay consistent with how `Remove` and `Set` have `recursiveRemove` and `recursiveSet` as well.